### PR TITLE
fix: Add UpdateDevice callback for device profile update

### LINKED ISSUE #1347: Can not find the new device resource because of device cache inconsistency
# 🐞 Bug Report

### Affected Services [**REQUIRED**]

 The issue is located in: device services

### Is this a regression?

 No

### Description and Minimal Reproduction [**REQUIRED**]
1. Start running edgex services
  ```
  docker compose -f docker-compose-no-secty.yml up -d
  ```
2. Create a new device resource to device-virtual profile `Random-Boolean-Device`
  ```
  curl -ki -X POST http://localhost:59881/api/v2/deviceprofile/resource -d '[  
    {
      "apiVersion": "v2","profileName":"Random-Boolean-Device",
      "resource": {
        "name": "resource-add-1", 
        "properties": {"valueType": "Bool","readWrite": "RW"}}
    }
  ]'
  ```
3. Send GET command to read the new device resource and return error
  ```
  curl -ki http://localhost:59900/api/v2/device/name/Random-Boolean-Device/resource-add-1   
  ```

  ```
  {"apiVersion":"v2","message":"error reading DeviceResource resource-add-1 for Random-Boolean-Device -\u003e resource not found","statusCode":500}
  ```
4. Restart device-virtual and send GET command again, return 200
  ```
  {"apiVersion":"v2","statusCode":200,"event":{"apiVersion":"v2","id":"713faa40-fa60-4b3d-aa06-c50675e953f5","deviceName":"Random-Boolean-Device","profileName":"Random-Boolean-Device","sourceName":"resource-add-1","origin":1677213763774591000,"readings":[{"id":"c6578840-16bf-4dd1-92db-5f6a6a69391b","origin":1677213763774591000,"deviceName":"Random-Boolean-Device","resourceName":"resource-add-1","profileName":"Random-Boolean-Device","valueType":"Bool","value":"true"}]}}
  ```

## 🔥 Exception or Error
<pre><code>

error reading DeviceResource resource-add-1 for Random-Boolean-Device -\u003e resource not found
</code></pre>

## 🌍  Your Environment
**Deployment Environment:**

**EdgeX Version [**REQUIRED**]:**
main branch/Leviski 

**Anything else relevant?**

### PR DESCRIPTION
**Pull request by jinlinGuan**
close #1347

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

1. build device-virtual from https://github.com/edgexfoundry/device-virtual-go/pull/347 based on this branch
2. create a new device resource
    ```
    curl -ki -X POST http://localhost:59881/api/v2/deviceprofile/resource -d '[  
      {
        "apiVersion": "v2","profileName":"Random-Boolean-Device",
        "resource": {
          "name": "resource-add-1", 
          "properties": {"valueType": "Bool","readWrite": "RW"}}
      }
    ]'
    ```
3. Send GET command to read the new device resource successfully
    ```
    curl -ki http://localhost:59900/api/v2/device/name/Random-Boolean-Device/resource-add-1
  
    {"apiVersion":"v2","statusCode":200,"event":{"apiVersion":"v2","id":"e59ecfae-f75f-4ea6-af03-05e2c7617119","deviceName":"Random-Boolean-Device","profileName":"Random-Boolean-Device","sourceName":"resource-add-1","origin":1677217694093349605,"readings":[{"id":"98ddb6e7-168f-46f1-a394-763d19393595","origin":1677217694093349605,"deviceName":"Random-Boolean-Device","resourceName":"resource-add-1","profileName":"Random-Boolean-Device","valueType":"Bool","value":"false"}]}}
    ```
4. Found `Invoked driver.UpdateDevice callback` in the logs
    ```
    level=DEBUG ts=2023-02-24T05:48:06.491984855Z app=device-virtual source=callback.go:58 msg="System event received on message queue. Topic: edgex/system-events/core-metadata/deviceprofile/update/device-virtual/Random-Boolean-Device, Correlation-id: 4004d944-2b6b-4292-8c54-7b0e6d57b8c6"
    level=DEBUG ts=2023-02-24T05:48:06.492308781Z app=device-virtual source=callback.go:39 msg="profile Random-Boolean-Device updated"
    level=DEBUG ts=2023-02-24T05:48:06.492389146Z app=device-virtual source=virtualdriver.go:133 msg="Device Random-Boolean-Device is updated"
    level=DEBUG ts=2023-02-24T05:48:06.492411647Z app=device-virtual source=callback.go:49 msg="Invoked driver.UpdateDevice callback for Random-Boolean-Device"
    ```

### CODE REVIEW

**Review by cloudxxx8 — DISMISSED** [review]
LGTM

File: internal/application/callback.go [hunk 1]
```diff
@@ -37,6 +37,19 @@ func UpdateProfile(profileRequest requests.DeviceProfileRequest, dic *di.Contain
 	}
 
 	lc.Debug(fmt.Sprintf("profile %s updated", profileRequest.Profile.Name))
```
Comment (cloudxxx8) [inline internal/application/callback.go:39]: please help update this line to use Debugf

Comment (jinlinGuan) [reply internal/application/callback.go]: Updated

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**